perf-site-replication.go

```diff
@@ -63,3 +67,38 @@ func (adm *AdminClient) SiteReplicationPerf(ctx context.Context, duration time.D
 	err = json.NewDecoder(resp.Body).Decode(&result)
 	return result, err
 }
+
+// SiteNetPerfReader - for TTFB and TotalResponseTime reader (4K stream)
+type SiteNetPerfReader struct {
+	TimeToFirstByte time.Duration
+	buf             []byte
+	startTime       time.Time
+	endTime         time.Time
+	readCounter     uint64
+}
+
+func (s *SiteNetPerfReader) Start() {
+	buf := make([]byte, 4*1024-1)
+	rand.Read(buf)
+	s.buf = buf
+	s.startTime = time.Now()
+}
+
+func (s *SiteNetPerfReader) Read(p []byte) (n int, err error) {
+	s.readCounter++
+	if s.readCounter == 1 {
+		n = copy(p, []byte{1})
+	} else if s.readCounter == 2 {
+		s.TimeToFirstByte = time.Now().Sub(s.startTime)
+		n = copy(p, s.buf)
+	} else {
+		return 0, io.EOF
```

Comment:
You do not need to only send the first byte via this reader instead it must be the first Read() itself how much was received by the other end. 

Look at the server-side implementation of how a response recorder is used to calculate time to the first byte.